sdk/core/core-http/lib/atomError.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { Constants } from "./util/constants";
+import { HttpOperationResponse } from "./httpOperationResponse";
+
+/**
+ * Represents custom `Error` information returned by ATOM based services.
+ */
+export class AtomError extends Error {
+  code?: string;
+  statusCode?: number;
+  additionalProperties?: any;
+  requestId?: string;
```

Comment:
As per offline discussions,
- Use RestError instead of AtomError
- Lift `message` and `code` from the error body in the constructor of the `RestError` 
- The `response` & `body` on the RestError will have all the information coming from the service, but we will not be pulling the dynamic properties up to `additionalProperties`